### ISSUE TITLE
Create a Method Explorer like in Dophin Smalltalk

### ISSUE THREAD

**Issue by brunobuzzi** [jgfoster/Jade#66]
Ok, i'm working on it.

**Comment by brunobuzzi** [comment]
![imagen](https://cloud.githubusercontent.com/assets/4719282/23187067/3eac6606-f868-11e6-82cc-cabee789d4b8.png)

**Comment by brunobuzzi** [comment]
It is possible to add nodes to the TreePresenter and drag&drop methods to these nodes.

![imagen](https://cloud.githubusercontent.com/assets/4719282/23231645/52bd5fb4-f927-11e6-8313-5b07810da970.png)

**Comment by jgfoster** [comment]
Yes, quite possible. We do something similar when dragging a method onto a class. See the "drag & drop" method category in JadeSystemBrowserPresenter.

**Comment by brunobuzzi** [comment]
Sorry for my english :) it was rather a report than a question !

**Comment by brunobuzzi** [comment]
Almost done ! Needs a little testing !
![imagen](https://cloud.githubusercontent.com/assets/4719282/23273737/f4b7c7f2-f9de-11e6-82e1-f990cf29f37b.png)

**Comment by jgfoster** [comment]
Actually your English was correct ("It is possible ..."); it was my reading that was incorrect ("Is it possible...?"). Now that I've merged this could you add some instructions on how to enable it?

**Comment by brunobuzzi** [comment]
The correct solution will be add a class JadeEnvironmentOptions and aGUI for the class. 
But for now just open implementors or references to some method. A regular Method Browser will be open, go to Menu Bar and choose Switch to ...

https://cloud.githubusercontent.com/assets/4719282/23344234/7b5bfaf6-fc57-11e6-86a0-de41991c9cff.png

**Comment by jgfoster** [comment]
When I ask to browse senders/implementors from within the list view (old style) it replaces the old window rather than opening a new one. I can see that adding to the existing tree view is appropriate, but I don't want to lose the original window when I ask for another list.

**Comment by brunobuzzi** [comment]
The best thing to do is to add a toggle toolbar button. If checked new senders&implementors browsing operations will a open new browser if not checked then open it inside the tree view.
What do you think ?

**Comment by jgfoster** [comment]
It seems to me that if we have two types of method list browsers, then having separate behavior is appropriate. If we have a simple list (the legacy implementation), then a new request should always open a new window (shell). If we have a tree list (the new implementation), then a new request should always add to the existing window (shell). 

The existing problem is that the introduction of the tree view has broken the legacy list view.

**Comment by brunobuzzi** [comment]
But the first Method Explorer opened from a System Browser should be the new implementation or the legacy implementation ?

**Comment by jgfoster** [comment]
Given that one can change the behavior I don't care much which is the initial selection. I think going with the new implementation would be nice.

**Comment by jgfoster** [comment]
I've reverted to the legacy implementation for now and fixed a couple issues. I'm happy to have the new tree view brought back and I'll try to test it better when it comes.
